src/Node/Render.php

```diff
@@ -51,7 +51,14 @@ protected function setEnvironment(\Twig_Environment $env) {
   }
 
   public static function getEnvironment(): \Twig_Environment {
-    return static::$env;
+    $env = static::$env;
+    if (!$env) {
+      $loader = \Drupal::service('twig.loader.twig_fractal');
+      $env = new \Twig\Environment($loader);
```

Comment:
Please revert this. The module is not supposed to create a new twig environment. That needs to be done in the wrapping application which uses the module/package.

Reply:
Right, this was an issue I had to pick when 'render' worked as twig 'include' tag without reinitiating the context for the included component and just using the same context on all rendered blocks. It appeared to be like an issue sorted within the module itself.

> That needs to be done in the wrapping application which uses the module/package.
Do you have a suggestion of how this can be controlled at the application level?

Reply:
jsut fyi more info on the the error thrown when the twig env is not reset

```
The website encountered an unexpected error. Please try again later.
TypeError: Return value of Drupal\twig_fractal\Node\Render::getEnvironment() must be an instance of Twig\Environment, null returned in Drupal\twig_fractal\Node\Render::getEnvironment() (line 56 of modules/contrib/twig_fractal/src/Node/Render.php).
Drupal\twig_fractal\Node\Render::getEnvironment() (Line: 67)
```

Reply:
Sorry, I was confused here.  The Render tag does this on its own when compiling the template: https://github.com/netzstrategen/twig_fractal/blob/bf355130e411953924e87f3ee064e131de5055b4/src/Node/Render.php#L71

Reply:
for some reason, `$compiler->getEnvironment()` seems to be null after first load. When the pages with patterns are loaded first time in Drupal, twig is loaded fine and rendering the patterns as expected but in the second load, it throws up the error that I mentioned above stating` Drupal\twig_fractal\Node\Render::getEnvironment() must be an instance of Twig\Environment, null returned`. 
After clearing cache from Drupal then Twig env is back and again in the second load being null.

should there be a cache handler setup to store the Twig env state?

Reply:
@sun just realised this has been already added as an issue by another user here - https://github.com/netzstrategen/twig_fractal/issues/26